core-runner/src/ext.rs

```diff
@@ -53,14 +54,43 @@ impl Ext {
 }
 
 impl EnvExt for Ext {
-    fn alloc(&mut self, pages: PageNumber) -> Result<PageNumber, &'static str> {
-        self.gas(pages.raw() * self.alloc_cost as u32)?;
+    fn alloc(&mut self, pages_num: PageNumber) -> Result<PageNumber, &'static str> {
+        // Greedly sub gas for allocations
+        self.gas(pages_num.raw() * self.alloc_cost as u32)?;
+        // Greedly sub gas for grow
```

Comment:
Is greedy subtraction necessary here? It may lead to unintuitive cases where program is given just enough gas to complete, but fails if it does not have enough gas for greedy subtraction.

also minor: `Greedly` -> `Greedily`
or
`Greedy gas subtraction for ...`

Reply:
It's not necessary, but it's the least difficult way. We cannot charge gas after allocation, because we must charge before action. So we can also charge for each grow and for each new allocated page separately, but this solution will be much bigger.